src/WSServerCon.cpp

```diff
@@ -596,6 +596,11 @@ void WSServerCon::processMessage(const QString &message)
         }
         );
     }
+    else if (root["msg"] == "lock_device")
+    {
+        const auto cb = [](bool, const QByteArray &, bool &){};
+        mpdevice->sendData(MPCmd::Command::LOCK_DEVICE, cb);
```

Comment:
This is not what you want to do. You need to add a new function in MPDevice that will create a new AsyncJob and put it onto the job queue. Directly sending a data packet like that will bypass the current running queue of command and can mess with the communication at some point

Reply:
That makes sense. Will change it into that approach.